### PR TITLE
feat: meru improved model picker

### LINKED ISSUE #3872: [Meru AB] Instructions screen - Update Model menu in advanced settings
See figma [here](https://www.figma.com/file/QOxHwppG64GtPocpDhS6Y7/Product?type=design&node-id=4758-20748&mode=design&t=CrHO9TDk6FhlGRRq-4) for new menu

The current menu is like this

Image

### PR DESCRIPTION
**Pull request by fontanierh**
## Description

fixes https://github.com/dust-tt/dust/issues/3872

I invented some descriptions, probably not the ones we want

Screenshot 2024-02-22 at 12 05 48

### CODE REVIEW

File: front/components/assistant_builder/AssistantBuilder.tsx [hunk 1]
```diff
@@ -182,6 +187,28 @@ const CREATIVITY_LEVELS = [
...
+const GPT_4_SHORT_DESCRIPTION = "Most polyvalent model";
+
+const MODEL_SHORT_DESCRIPTIONS: Record<ModelId, string> = {
+  "gpt-3.5-turbo-1106": "OpenAI's fastest model.",
+  "gpt-4-turbo-preview": GPT_4_SHORT_DESCRIPTION,
+  "gpt-4": GPT_4_SHORT_DESCRIPTION,
+  "gemini-pro": "Scales across a wide range of tasks.",
+  "mistral-small": "Very fast and effective.",
+  "mistral-medium": "One of the most powerful OSS model.",
+  "mistral-next": "Le gros modèle.",
```
[the remark this answers is not in the document]
Comment (spolu) [reply front/components/assistant_builder/AssistantBuilder.tsx:208]: Let's have these short desciptions live with the long ones on the model cards

[the remark this answers is not in the document]
Comment (spolu) [reply front/components/assistant_builder/AssistantBuilder.tsx:208]: I think we don't want to qualify models here even if that would be the goal but this is ~ mission impossible.

I would propose to be very pragmatic

GPT3.5
OpenAI's fast model

GPT4
OpenAI's smartest model

Claude 2.1
Anthropic's smartest model

Claude Instant 1.2
Anthropic fast model

Mistral Small
Mistral fast model

Mistral Medium
Mistral smartest model

Mistral (next)
Next generation Mistral model

Gemini Pro
Google smartest model

[the remark this answers is not in the document]
Comment (fontanierh) [reply front/components/assistant_builder/AssistantBuilder.tsx:208]: thanks, will use that 🙏  tried to do a bit like the figma but mission impossible indeed and feels quite wrong too

**Comment by spolu** [comment]
As per IRL let's centralize the description + left some suggestions above

**Review by spolu — APPROVED** [review]

**Pull request merged**